client/consensus/aura/src/import_queue.rs

```diff
@@ -186,16 +186,15 @@ where
 		&mut self,
 		mut block: BlockImportParams<B, ()>,
 	) -> Result<(BlockImportParams<B, ()>, Option<Vec<(CacheKeyId, Vec<u8>)>>), String> {
-		// When importing whole state we don't verify the seal as the state is not available.
-		if block.with_state() {
-			return Ok((block, Default::default()))
-		}
-
-		// Skip checks that include execution, if being told so.
+		// Skip checks that include execution, if being told so or when importing only state.
 		//
 		// This is done for example when gap syncing and it is expected that the block after the gap
 		// was checked/chosen properly, e.g. by warp syncing to this block using a finality proof.
-		if block.state_action.skip_execution_checks() {
+		// Or when we are importing state only and can not verify the seal.
+		if block.with_state() || block.state_action.skip_execution_checks() {
+			// When we are importing only the state of a block, it will be the best block.
+			block.fork_choice = Some(ForkChoiceStrategy::Custom(block.with_state()));
```

Comment:
Another observation.
Probably this requirement to modify the `fork_choice` is common to other protocols as well (for sure Babe).

Babe currently perform this a bit later [here](https://github.com/paritytech/substrate/blob/f4bc99303e6faa0ae0461b4606c1699b4463ff4b/client/consensus/babe/src/lib.rs#L1377). In the `import_state`.

 Can't we move this assignment to the `block` params construction in the import queue? (i.e. close to [this](https://github.com/paritytech/substrate/blob/f4bc99303e6faa0ae0461b4606c1699b4463ff4b/client/consensus/common/src/import_queue.rs#L326) so that we catch them all in one single place?

If this setting of fork choin

Reply:
I think the cumulus one needs to stay anyway, as it is unrelated to all the state-business.

Reply:
I see. I was just trying to find out if there is a way to have these less scattered around. 
That is, probably babe and aura are similar WRT this aspect. But aura is setting it in the verifier while the other in the `block_import->state_import`